### PR TITLE
Support rotated HGCal Wafers in Fireworks

### PR DESCRIPTION
**Pull request by rovere**
#### PR description:

This PR addresses the following point:

* Improve geometry handling in Fireworks by supporting fully rotated layers in HGCal. The old handling of reconstruction geometry in Fireworks was too decoupled from the tools developed to create and handle the HGCal geometry. Now every aspect of the geometry relevant for a correct rendering is handled while creating the geometry itself, not by homemade code at rendering time. Comments in the single git commits and inside the code are far more explanatory.
* Correctly handle the cells' orientation of the fully rotated layers. The improved handling should be generic enough to be valid for arbitrary rotations, not just the current one at 30 degrees. Comments in the single git commits and inside the code contain more information.
* Correctly pass the thickness of every single cell or tile inside the geometry, so that the `3DRecHits` rendering of the single cells and tiles is properly done. Comments have been added to the code to alert the developer about the downstream effects in case she wants to change the behaviour.
* Update the Fireworks plugins to use the feature that has been added and stored along with the HGCal reconstruction geometry.

#### PR validation:

The code has been tested using `D86` and `D88` geometry. No further bugs nor strange behaviour is observed.
In particular, layers 28, 30 and 32 show the correct rendering, as shown in the following images:
Layer28_D86
Layer30_D86
Layer32_D86

Moreover, one can see the different orientations of the single cells within the rotated layers with respect to the default orientation, as shown in this picture:

cellsOrientation_28_30_32

### CODE REVIEW

**Comment by rovere** [comment]
@cmsbuild please check

File: Geometry/HGCalGeometry/src/HGCalGeometry.cc [hunk 1]
```diff
@@ -402,7 +402,8 @@ HGCalGeometry::CornersVec HGCalGeometry::getNewCorners(const DetId& detid, bool
       co[i] = GlobalPoint(
           (r + signr[i] * dr) * cos(fi + signf[i] * dfi), (r + signr[i] * dr) * sin(fi + signf[i] * dfi), (v.z() + dz));
     }
-    co[ncorner - 1] = co[0];
+    // Used to pass downstream the thickness of this cell
+    co[ncorner - 1] = GlobalPoint(0, 0, -2 * dz);
```
Comment (srimanob) [inline Geometry/HGCalGeometry/src/HGCalGeometry.cc:406]: Would you mind to comment on this line? Do you expect something to change from this line? Thx.

**Comment by cvuosalo** [comment]
+1

**Comment by alja** [comment]
+1

File: Geometry/HGCalGeometry/src/HGCalGeometry.cc (hunk 1 above)
Comment (bsunanda) [reply Geometry/HGCalGeometry/src/HGCalGeometry.cc:406]: This is an understanding between geometry and fireworks - in addition to the corners fireworks also needs half the width. Marco has correctly put it in.

**Comment by bsunanda** [comment]
@srimanob Could you please approve this PR. Needed to define V17 geometry

**Comment by srimanob** [comment]
+Upgrade

This PR is to update geometry related to the firework on HGCal.

**Pull request merged**